File: frontend/src/api/product.ts
import { http } from './http';

export interface ProductPayload {
  name: string;
  description: string;
  price?: number;
  imageUrl?: string;
  status?: 'ACTIVE' | 'DISABLED' | 'OUT_OF_STOCK';
}

export const productApi = {
  list() {
    return http.get<ProductPayload[]>('/products');
  },
  get(id: number) {
    return http.get<ProductPayload>(`/products/${id}`);
  },
  create(payload: ProductPayload) {
    return http.post<ProductPayload>('/products', payload);
  },
  update(id: number, payload: ProductPayload) {
    return http.put<ProductPayload>(`/products/${id}`, payload);
  },
  remove(id: number) {
    return http.delete<void>(`/products/${id}`);
  },
};

File: backend/src/main/java/com/ainativecms/entity/Product.java
package com.ainativecms.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;

@Entity
@Table(name = "products")
public class Product extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    private Double price;

    private String imageUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status = Status.ACTIVE;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public enum Status {
        ACTIVE,
        DISABLED,
        OUT_OF_STOCK
    }
}
